website/pages/docs/platform/k8s/helm/examples/enterprise-best-practice.mdx

```diff
@@ -0,0 +1,133 @@
+assume you name this new role, "EKSNodeGroupForAutoUnseal".
+3. While the cluster is launching, also create a KMS key that will be used for auto-unseal.
+Under "Define key usage permissions", search for "EKSNodeGroupForAutoUnseal" and add a check
+next to it.
+4. One you've finished creating the key, return to IAM. Create a new policy called
+"UnsealKeyUsageForVault" and attach it to the "EKSNodeGroupForAutoUnseal" role. The policy
+contents should be:
+
+```
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "kms:Decrypt",
+                "kms:Encrypt",
+                "kms:DescribeKey"
+            ],
+            "Resource": "arn:aws:kms:us-west-2:012345678901:key/207a68f5-f100-410c-8066-EXAMPLE"
+        }
+    ]
+}
+```
+
+5. Return to EKS and add a node group. While launching it, under "Node IAM Role Name", select
+"EKSNodeGroupForAutoUnseal".
+6. While the nodes are creating, point your local `kubectl` at your new cluster as shown
+[here](https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html).
+
+## Generate TLS Certificates
+
+TLS certificate generation is covered [here](/docs/platform/k8s/helm/examples/standalone-tls).
+
+## Setting Up the Primary Cluster
+
+The below `values.yaml` can be used to set up a three server Vault cluster using
+Consul as a highly available storage backend, with AWS KMS for Auto Unseal.
+
+```yaml
+server:
+  image:
+    repository: "hashicorp/vault-enterprise"
+    tag: "1.3.1_ent"
+
+  affinity: |
+    podAntiAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        - labelSelector:
+            matchLabels:
+              app: {{ template "vault.name" . }}
+              release: "{{ .Release.Name }}"
+              component: server
+          topologyKey: kubernetes.io/hostname
+
+  service:
+    enabled: true
+
+  ha:
+    enabled: true
+    replicas: 3
+
+    config: |
+      ui = true
+
+      listener "tcp" {
+        tls_cert_file = "/path/to/fullchain.pem"
+        tls_key_file  = "/path/to/privkey.pem"
+        tls_client_ca_file = "/vault/userconfig/vault-server-tls/vault.ca"
+        address = "[::]:8200"
+        cluster_address = "[::]:8201"
+      }
+
+      storage "consul" {
+        path = "vault"
+        address = "HOST_IP:8500"
+        token = "45a3bd52-07c7-47a4-52fd-0745e0cfe967"
+      }
+
+      seal "awskms" {
+        region     = "us-west-2"
+        kms_key_id = "19ec80b0-dfdd-4d97-8164-c6examplekey"
+      }
+```
+
+Note that in the above seal stanza, the AWS key and secret are omitted. That's because
+Vault picks them up from the node group's Instance Metadata, which in this example has
+been configured to provide credentials for the "EKSNodeGroupForAutoUnseal" role.
+
+## Setting Up a Disaster Replication (DR) Cluster
+
+Setting up a DR cluster follows an identical process and configuration for
```

Comment:
I think we need documentation here about using the `active` service when setting up DR/Performance clusters.  Additionally, this is a confusing process because you actually need to run `helm install` twice to get two clusters.  I think it would be helpful to show this workflow similar to what was done in the automated testing:

* https://github.com/hashicorp/vault-helm/blob/enterprise/test/acceptance/server-ha-enterprise.bats#L8
* https://github.com/hashicorp/vault-helm/blob/enterprise/test/acceptance/server-ha-enterprise.bats#L66

The active service name would go inside the token we generate to setup the cluster: https://github.com/hashicorp/vault-helm/blob/enterprise/test/acceptance/server-ha-enterprise.bats#L57

Reply:
I added this in https://github.com/hashicorp/vault/pull/8283/commits/da348bc439957bb0ccb1d9af6c7a580b61670598, and hopefully it synthesizes the data well enough for an end user, but let me know if you were thinking something different.

Reply:
This looks good!  Thank you!

Reply:
One last thing here: we need a note about using the active service to setup replication.

Reply:
Just want to note here that we chatted about this, and decided that further elaboration would be better suited to a Learn guide. To that effect, I have emailed our Education Team the details of what we'd love to see included in it.